kale/embed/video_se_i3d.py

```diff
@@ -0,0 +1,231 @@
+            model.Mixed_4d.add_module("SELayerMAC", SELayerMAC(512))
+            model.Mixed_4e.add_module("SELayerMAC", SELayerMAC(528))
+            model.Mixed_4f.add_module("SELayerMAC", SELayerMAC(832))
+            model.Mixed_5b.add_module("SELayerMAC", SELayerMAC(832))
+            model.Mixed_5c.add_module("SELayerMAC", SELayerMAC(1024))
+
+        elif attention == "SELayerCT":
+            model.Mixed_3b.add_module("SELayerCTc", SELayerC(256))
+            model.Mixed_3c.add_module("SELayerCTc", SELayerC(480))
+            model.Mixed_4b.add_module("SELayerCTc", SELayerC(512))
+            model.Mixed_4c.add_module("SELayerCTc", SELayerC(512))
+            model.Mixed_4d.add_module("SELayerCTc", SELayerC(512))
+            model.Mixed_4e.add_module("SELayerCTc", SELayerC(528))
+            model.Mixed_4f.add_module("SELayerCTc", SELayerC(832))
+            model.Mixed_5b.add_module("SELayerCTc", SELayerC(832))
+            model.Mixed_5c.add_module("SELayerCTc", SELayerC(1024))
+
+            model.Mixed_3b.add_module("SELayerCTt", SELayerT(n // 2))
+            model.Mixed_3c.add_module("SELayerCTt", SELayerT(n // 2))
+            model.Mixed_4b.add_module("SELayerCTt", SELayerT(n // 4))
+            model.Mixed_4c.add_module("SELayerCTt", SELayerT(n // 4))
+            model.Mixed_4d.add_module("SELayerCTt", SELayerT(n // 4))
+            model.Mixed_4e.add_module("SELayerCTt", SELayerT(n // 4))
+            model.Mixed_4f.add_module("SELayerCTt", SELayerT(n // 4))
+            # model.Mixed_5b.add_module("SELayerCTt", SELayerT(n // 8))
+            # model.Mixed_5c.add_module("SELayerCTt", SELayerT(n // 8))
+
+        elif attention == "SELayerTC":
+            model.Mixed_3b.add_module("SELayerTCt", SELayerT(n // 2))
+            model.Mixed_3c.add_module("SELayerTCt", SELayerT(n // 2))
+            model.Mixed_4b.add_module("SELayerTCt", SELayerT(n // 4))
+            model.Mixed_4c.add_module("SELayerTCt", SELayerT(n // 4))
+            model.Mixed_4d.add_module("SELayerTCt", SELayerT(n // 4))
+            model.Mixed_4e.add_module("SELayerTCt", SELayerT(n // 4))
+            model.Mixed_4f.add_module("SELayerTCt", SELayerT(n // 4))
+            model.Mixed_5b.add_module("SELayerTCt", SELayerT(n // 8))
+            model.Mixed_5c.add_module("SELayerTCt", SELayerT(n // 8))
+
+            model.Mixed_3b.add_module("SELayerTCc", SELayerC(256))
+            model.Mixed_3c.add_module("SELayerTCc", SELayerC(480))
+            model.Mixed_4b.add_module("SELayerTCc", SELayerC(512))
+            model.Mixed_4c.add_module("SELayerTCc", SELayerC(512))
+            model.Mixed_4d.add_module("SELayerTCc", SELayerC(512))
+            model.Mixed_4e.add_module("SELayerTCc", SELayerC(528))
+            model.Mixed_4f.add_module("SELayerTCc", SELayerC(832))
+            model.Mixed_5b.add_module("SELayerTCc", SELayerC(832))
+            model.Mixed_5c.add_module("SELayerTCc", SELayerC(1024))
```

Comment:
any redundancy can be reduced?

Reply:
I tried to search for a better way to add a new layer to the existing model, but I failed.

Reply:
Similar to the above comment, see where the redundancies are and whether this is a better way to write it more efficiently, e.g. those code for SELayerC, SELayerCoC, SELayerMC, SELayerMAC differ only by the layer name. It is possible  to make the layer name some functional variable like in Ada?

Anyway, these are more optimization, if stuck, you could put these notes as comments in the code for future optimization, possibly by others since it is a quite isolated issue. We do not have to be perfect on the first attempt.  This advice applies to other comments made as well. The review identified many issues, but some can be left for future to address.